app/src/main/java/com/android/shelfLife/ui/overview/ChooseFoodItem.kt

```diff
@@ -0,0 +1,284 @@
+import androidx.compose.foundation.layout.*
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.lazy.grid.GridCells
+import androidx.compose.foundation.lazy.grid.LazyVerticalGrid
+import androidx.compose.foundation.lazy.grid.items
+import androidx.compose.foundation.rememberScrollState
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.Upload
+import androidx.compose.material3.*
+import androidx.compose.runtime.*
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.graphicsLayer
+import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.unit.dp
+import androidx.hilt.navigation.compose.hiltViewModel
+import coil3.compose.AsyncImage
+import coil3.compose.rememberAsyncImagePainter
+import com.android.shelfLife.R
+import com.android.shelfLife.model.foodFacts.FoodFacts
+import com.android.shelfLife.model.foodFacts.FoodUnit
+import com.android.shelfLife.model.foodFacts.Quantity
+import com.android.shelfLife.model.foodFacts.SearchStatus
+import com.android.shelfLife.model.foodItem.FoodItem
+import com.android.shelfLife.ui.navigation.NavigationActions
+import com.android.shelfLife.ui.navigation.Route
+import com.android.shelfLife.ui.navigation.Screen
+import com.android.shelfLife.ui.utils.CustomButtons
+import com.android.shelfLife.ui.utils.CustomTopAppBar
+import com.android.shelfLife.viewmodel.camera.BarcodeScannerViewModel
+import com.android.shelfLife.viewmodel.overview.FoodItemViewModel
+import com.google.firebase.storage.FirebaseStorage
+import com.google.firebase.storage.StorageReference
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.tasks.await
+
+@Composable
+fun ChooseFoodItem(
+    navigationActions: NavigationActions,
+    foodItemViewModel: FoodItemViewModel = hiltViewModel(),
+    cameraViewModel: BarcodeScannerViewModel = hiltViewModel()
+) {
+  val coroutineScope = rememberCoroutineScope()
+  val context = LocalContext.current
+
+  val selectImageLauncher =
+      rememberLauncherForActivityResult(contract = ActivityResultContracts.GetContent()) { uri: Uri?
+        ->
+        foodItemViewModel.capturedImageUri = uri
+      }
+
+    LaunchedEffect(foodItemViewModel.capturedImageUri) {
+        foodItemViewModel.capturedImageUri?.let { uri ->
+            foodItemViewModel.isLoading = true
+            val uploadedUrl = foodItemViewModel.uploadImageToFirebaseStorage(uri, context)
+            if (uploadedUrl != null) {
+                foodItemViewModel.selectedImage = FoodFacts(
+                    name = "",
+                    quantity = Quantity(1.0, FoodUnit.GRAM),
```

Comment:
It'd be good to add some comments here to specify that this is a temporary food item that is used for the next screen